tests/memory_tests/src/memory_tests/memtest_infer.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <inference_engine.hpp>
+#include <iostream>
+
+#include "common.h"
+#include "memory_tests_helper/memory_counter.h"
+#include "memory_tests_helper/utils.h"
+using namespace InferenceEngine;
+
+
+/**
+ * @brief Function that contain executable pipeline which will be called from
+ * main(). The function should not throw any exceptions and responsible for
+ * handling it by itself.
+ */
+int runPipeline(const std::string &model, const std::string &device) {
+  auto pipeline = [](const std::string &model, const std::string &device) {
+    Core ie;
+    CNNNetwork cnnNetwork;
+    ExecutableNetwork exeNetwork;
+    InferRequest inferRequest;
+    size_t batchSize = 0;
+
+    {
+      SCOPED_MEM_COUNTER(full_run);
```

Comment:
No need to track scopes due memory snapshots is taken as is (no `finish - start` computations). Replace with MEMORY_SNAPSHOT macro